vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet_create.cc

```diff
@@ -6,83 +6,111 @@
 #include "bat/ledger/internal/wallet/wallet_create.h"
 
 #include <utility>
-#include <vector>
 
 #include "bat/ledger/internal/common/security_util.h"
 #include "bat/ledger/internal/common/time_util.h"
-#include "bat/ledger/internal/constants.h"
+#include "bat/ledger/internal/endpoints/patch_wallets/patch_wallets.h"
+#include "bat/ledger/internal/endpoints/post_wallets/post_wallets.h"
+#include "bat/ledger/internal/endpoints/request_for.h"
 #include "bat/ledger/internal/ledger_impl.h"
 #include "bat/ledger/internal/logging/event_log_keys.h"
 
-using std::placeholders::_1;
-using std::placeholders::_2;
+using ledger::endpoints::PatchWallets;
+using ledger::endpoints::PostWallets;
+using ledger::endpoints::RequestFor;
 
-namespace ledger {
-namespace wallet {
+namespace ledger::wallet {
 
-WalletCreate::WalletCreate(LedgerImpl* ledger) :
-    ledger_(ledger),
-    promotion_server_(std::make_unique<endpoint::PromotionServer>(ledger)) {
+WalletCreate::WalletCreate(LedgerImpl* ledger) : ledger_(ledger) {
   DCHECK(ledger_);
 }
 
-WalletCreate::~WalletCreate() = default;
-
-void WalletCreate::Start(ledger::ResultCallback callback) {
+void WalletCreate::CreateWallet(absl::optional<std::string>&& geo_country,
```

Comment:
I can see how this can be confusing, so renaming it back to `Start()`. `WalletCreate` and `WalletRecover` are unnecessary abstractions. Instead, `Wallet` itself should have functions called `Create()` and `Recover()`. Will address these in a follow-up PR.